### PR TITLE
Refactoring of texture-tester app

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
Tester app is ready to be used.

image

### CODE REVIEW

File: examples/textures-tester/app.js [hunk 1]
```diff
@@ -1,157 +1,273 @@
...
+    this.atcSupported = false;
+    this.etc1Supported = false;
+    this.loadOptions = {basis: {}, image: {decode: true, type: 'image'}};
+  }
 
-const gl = canvas.getContext('webgl');
+  async renderTextures() {
+    const canvas = this.setupCanvas();
+    const gl = canvas.getContext('webgl');
+    this.setupSupportedForamts(gl);
```
Comment (ibgreen) [inline examples/textures-tester/app.js:35]: Type formats

File: examples/textures-tester/app.js [hunk 2]
```diff
@@ -1,157 +1,273 @@
...
-  try {
-    const {width, height, compressed, format, data} = image;
-    if (compressed) {
-      gl.compressedTexImage2D(gl.TEXTURE_2D, 0, format, width, height, 0, data);
-    } else {
-      gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, image);
+  async loadTestTextures(canvas, gl, loadOptions, program) {
+    const textures = document.querySelectorAll('.texture');
+
+    for (let index = 0; index < textures.length; index++) {
```
Comment (ibgreen) [inline examples/textures-tester/app.js:118]: `for const texture of textures`?

**Review by ibgreen — APPROVED** [review]

File: examples/textures-tester/app.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply examples/textures-tester/app.js]: Fixed

File: examples/textures-tester/app.js (hunk 2 above)
Comment (dryabinin-actionengine) [reply examples/textures-tester/app.js]: Fixed

**Pull request merged**